packages/flutter_tools/test/integration/lifetime_test.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:test/test.dart';
+
+import 'test_data/basic_project.dart';
+import 'test_driver.dart';
+
+BasicProject _project = new BasicProject();
+FlutterTestDriver _flutter;
+
+const Duration requiredLifespan = Duration(seconds: 5);
+
+void main() {
+  group('flutter run', () {
+    setUp(() async {
+      final Directory tempDir = await fs.systemTempDirectory.createTemp('test_app');
+      await _project.setUpIn(tempDir);
+      _flutter = new FlutterTestDriver(tempDir);
+    });
+
+    tearDown(() async {
+      try {
+        await _flutter.stop();
+        _project.cleanup();
+      } catch (e) {
+        // Don't fail tests if we failed to clean up temp folder.
```

Comment:
This seems a bit fishy.
Failing to clean up seems to indicate that something went wrong.
At least we should check on the exception type to see it is as expected for a failed cleanup.
Also the try should only span `_project.cleanup()` and we could also write a message indicating that cleanup failed.

Reply:
This was because sending `app.stop` could fail depending on the state of the app (eg. it's not valid if it didn't launch); but I think that's not so valid anymore (firstly, if the test has failed anyway, it doesn't matter if the throw, and secondly there have been many improvements to `stop()` in other PRs to make it more resilient).

I'll remove it here and we'll see how it goes on the bots. I've also made a note to repeat in the other integration tests (but will do in another PR).